subsys/net/lib/sockets/sockets.c

```diff
@@ -353,6 +357,7 @@ Z_SYSCALL_HANDLER(zsock_connect, sock, addr, addrlen)
 	return z_impl_zsock_connect(sock, (struct sockaddr *)&dest_addr_copy,
 				   addrlen);
 }
+#include <syscalls/zsock_connect_mrsh.c>
```

Comment:
Just a quick question, as I'm still reviewing this and haven't wrapped my head completely around how the build in this new scheme works. Instead of having to #include a separate C file for every syscall implemented, would it be possible to just put all the syscall marshalling functions in a single C file which gets built and linked in? or is there something specific which requires each marshalling function to be included in this way?

Reply:
That's the way the original scheme worked, actually (they were in syscall_dispatch.c).  But it has the side effect of being unable to inline the impl function unless it is in a header.  This way they can live wherever the user wants them.

Personally I don't think that extra function call per actually-taken-syscall is a big deal, so I'd be happy with either scheme.

Reply:
> Personally I don't think that extra function call per actually-taken-syscall is a big deal, so I'd be happy with either scheme.

If I understand you correctly I am fine with that also and would rather just put them all in a single C dispatch file.

The inlining I was worried about was supervisor mode calls to syscall APIs, which are currently inlined if the z_impl_xxxx() function for it is declared inline.

Reply:
+1 to putting the marshalling functions to separate C file.

Also, why not use the macro to create the function name so instead of this
```
static inline int z_vrfy_zsock_listen(int sock, int backlog)
```
we could have this
```
int Z_SYSCALL_HANDLER(zsock_listen, int sock, int backlog)
```
so user does not need to know / remember how the function should be named.